### DISCUSSION TITLE
安卓使用s3连接报错：Error: Request Failed. SSLHandshakeException java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.

### DISCUSSION THREAD

**Discussion by youarenever** [remotely-save/remotely-save#672] [Q&A]
使用自己nas中的s3服务，ssl使用的let's encrypt的免费证书。pc端和ios端均连接正常，且能正常同步。
但是在安卓端连接就会报错
错误信息：
Error: Request Failed. SSLHandshakeException java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.

我搜索相关相关信息后，按照某些方法，把let's encrypt的证书手动安装到安卓上，但仍然无法连接。
请问有没有什么解决办法？

**Comment by fyears** [comment]
看一下是否可以解决呢 fullchain.pem 和 chain.pem 都试一下。

https://stackoverflow.com/questions/45488917/lets-encrypt-on-android-gives-java-security-cert-certpathvalidatorexception-tr

此外也可以换 zerossl 试试

**Reply by youarenever** [reply]
排查下来，应该是这个服务的ca证书的问题，但我用的威联通自带的s3服务和自动申请的ssl证书，改证书不太方便。
最后只能同时设置了http和https，安卓暂时就用http。
感谢！
